### PR TITLE
[DROOLS-7635] ansible-rulebook : Raise an error when a condition comp…

### PR DESCRIPTION
**Pull request by tkobayas**
…ares incompatible types

- alpha index test, beta index test

### CODE REVIEW

**Comment by tkobayas** [comment]
Added test cases:
- alphaIndex : Not match when type mismatch (expected). No error log (limitation)
- betaIndex : Not match when type mismatch (expected). No error log (limitation)
- negation : Not match when type mismatch (expected). Error log produced (expected)
- any: Match when type mismatch on one condition, but properly match the other condition (expected). Error log produced (expected)

File: drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/NegationOperator.java [hunk 1]
```diff
@@ -14,6 +16,14 @@ public NegationOperator(ConstraintOperator toBeNegated) {
 
     @Override
     public <T, V> BiPredicate<T, V> asPredicate() {
+        if (toBeNegated instanceof RulebookConstraintOperator) {
+            return (t, v) -> {
+                // if not compatible type, return false regardless of the result
+                // but let the operator be executed anyway so that the error message specific to the constraint is logged
+                boolean result = !toBeNegated.asPredicate().test(t, v);
+                return isCompatibleType(t, v) ? result : false;
```
Comment (mariofusco) [inline drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/NegationOperator.java:24]: Minor performance improvement: isn't necessary to compute the result if types aren't compatible.

**Review by mariofusco — APPROVED** [review]
Ok for me, I just added a suggestion for a minor improvement.

File: drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/NegationOperator.java (hunk 1 above)
Comment (tkobayas) [reply drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/NegationOperator.java]: I wanted to let the operator log the type check warning. But yes, I can refactor it to make it cleaner.

**Pull request merged**